source/extensions/tracers/common/ot/opentracing_driver_impl.cc

```diff
@@ -21,52 +21,57 @@ Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::Request
     ot_span_context_handle(Http::CustomHeaders::get().OtSpanContext);
 
 namespace {
-class OpenTracingHTTPHeadersWriter : public opentracing::HTTPHeadersWriter {
+class OpenTracingHeadersWriter : public opentracing::HTTPHeadersWriter {
```

Comment:
🤔 I think there is no problem, because here only simple set and get operations are needed on TraceContext. So it is completely possible to use TraceContext to replace HeaderMap.
TraceContext does not support iterative API for the time being, so dynamic_cast is used to temporarily bypass this problem.
And this is an overall change, we can't just keep this one class without making changes.

Reply:
The superclass is HTTPHeadersWriter so this is still HTTP only?

Reply:
@lizan Apart from the name, `HTTPHeadersWriter` does not seem to have anything related to the HTTP protocol. 

I have checked source code of opentracing interface and also implementation of `datadog/lightstep`. `HTTPHeadersWriter` does not contain any HTTP-dependent interfaces. `HTTPHeadersWriter` simply inherits from `TextMapWriter`, without adding any HTTP-related methods. 

```
class TextMapReader {
 public:
  virtual ~TextMapReader() = default;
  virtual expected<string_view> LookupKey(string_view /*key*/) const {
    return make_unexpected(lookup_key_not_supported_error);
  }
  virtual expected<void> ForeachKey(
      std::function<expected<void>(string_view key, string_view value)> f)
      const = 0;
};

class TextMapWriter {
 public:
  virtual ~TextMapWriter() = default;
  virtual expected<void> Set(string_view key, string_view value) const = 0;
};

class HTTPHeadersReader : public TextMapReader {};
class HTTPHeadersWriter : public TextMapWriter {};
```
`datadog/lightstep` also does not provide any operations for `HTTPHeadersWriter` that are different from `TextMapWriter`.

In summary, `HTTPHeadersWriter` will not affect the versatility of tracers.
We can completely replace it with `TextMapWriter`. The reason why I have not done this at present is because I don't want the general tracing system to have any impact on existing HTTP Tracers users.